msal/src/main/java/com/microsoft/identity/client/internal/MsalUtils.java

```diff
@@ -259,12 +280,19 @@ public static boolean hasCustomTabRedirectActivity(@NonNull final Context contex
         for (final ResolveInfo info : resolveInfoList) {
             final ActivityInfo activityInfo = info.activityInfo;
 
-            if (activityInfo.name.equals(BrowserTabActivity.class.getName())) {
+            if (activityInfo.name.equals(BrowserTabActivity.class.getName()) &&
+                    activityInfo.packageName.equals(context.getPackageName())) {
                 hasActivity = true;
             } else {
                 // another application is listening for this url scheme, don't open
                 // Custom Tab for security reasons
-                return false;
+                Logger.warn(TAG, "Another application is listening for the URL scheme. " +
+                        "This app's package name is " + activityInfo.packageName);
+                throw new MsalClientException(
```

Comment:
We don't seem to do anything with this exception, so I don't see much value in throwing it.

Reply:
That's right...we are throwing it...all the way to the caller

Reply:
The caller will receive it in `onError` callback

Reply:
This suggests otherwise... line 243
try {
            return validateCustomTabRedirectActivity(context, url);
        } catch (final MsalClientException e) {
            return false;
        }

Reply:
That is the API I deprecated (`hasCustomTabRedirectActivity` vs `validateCustomTabRedirectActivity`) ...I don't want to modify the behavior of the existing API as it is Public API...or we could choose to do it...I think this wasn't meant to be called by customers so not sure why it is Public...but now that it was public..I chose not to change the behavior

Reply:
I changed it back to the original code in the deprecated method and opted to have code duplication instead so we can easily remove it in the future. Addressed here: 91fe570